Description: make README compliant with Debian systems
Author: Joao Eriberto Mota Filho <eriberto@debian.org>
Forwarded: not-needed
Last-Update: 2020-11-07
Index: disktype/README
===================================================================
--- disktype.orig/README
+++ disktype/README
@@ -60,7 +60,8 @@ File systems: FAT12/FAT16/FAT32, NTFS, H
   cramfs, Linux squashfs, UFS (some variations), SysV FS (some
   variations), JFS, XFS, Amiga FS/FFS, Amiga SFS, Amiga PFS, BeOS BFS,
   QNX4 FS, UDF, 3DO CD-ROM file system, Veritas VxFS, Xbox DVD file
-  system.
+  system. On Debian systems, Btrfs, exFAT, Ext4 and F2FS are also
+  known.
 
 Partitioning: DOS/PC style, EFI GPT, Apple, Amiga "Rigid Disk", ATARI
   ST (AHDI3), BSD disklabel, Linux RAID physical disks, Linux LVM1
