---
 prog/detect/sensors-detect |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/prog/detect/sensors-detect
+++ b/prog/detect/sensors-detect
@@ -2679,8 +2679,8 @@
 sub initialize_kernel_version
 {
 	chomp($kernel_version = `uname -r`);
-	$kernel_version =~ /(\d+)\.(\d+)\.(\d+)(.*)/;
-	@kernel_version = ($1, $2, $3, $4);
+	$kernel_version =~ /(\d+)\.(\d+)(?:\.(\d+))?(.*)/;
+	@kernel_version = ($1, $2, $3 || 0, $4);
 	chomp($kernel_arch = `uname -m`);
 
 	# We only support kernels >= 2.6.5
