diff -Naur gftp-20210407/lib/sshv2.c gftp-e8d6f07e74e375589370136c082439899d4c78d7/lib/sshv2.c
--- gftp-20210407/lib/sshv2.c	2021-04-07 16:03:14.000000000 +0800
+++ gftp-e8d6f07e74e375589370136c082439899d4c78d7/lib/sshv2.c	2021-04-16 21:11:56.000000000 +0800
@@ -500,7 +500,8 @@
               break;
             }
         }
-      else if (strstr (tempstr, "(yes/no)?") != NULL)
+      else if (strstr (tempstr, "(yes/no)?") != NULL ||
+               strstr (tempstr, "(yes/no/[") != NULL)
         {
           clear_tempstr = 1;
           if (!gftpui_protocol_ask_yes_no (request, request->hostname, tempstr))
